File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {

    var answer = [];
    let word_stack = []
    s = s.split("")
    
    for ( let w of s) {
        if ( word_stack.includes(w) === false) {
            word_stack.push(w)
            answer.push(-1)
        } else {
            let idx = 1
            for (let i = word_stack.length -1 ; i > -1 ; i --) {
                if (word_stack[i] === w) {
                    word_stack.push(w)
                    answer.push(idx)
                    break
                } else {
                    idx += 1
                }
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    for i in range(len(s)):                #해당 문자열 길이만 큼 반복
        if s[i] in s[:i]:                  #해당 문자가 현재 인덱스까지의 문자열에 존재한다면
            while s[:i].count(s[i])>1:     #현재 인덱스까지의 문자열내에 여러개 존재한다면 1개가 남을때까지 반복
                s = s.replace(s[i],'0',1)  #해당 문자 replace
            answer.append(i-s.index(s[i])) #가까운 해당 문자의 길이 출력
        else:                              #해당 문자가 현재 인덱스까지의 문자열에 존재하지 않는다면
            answer.append(-1)              #-1 출력
    return answer